homeassistant/components/climate/melissa.py

```diff
@@ -0,0 +1,248 @@
+"""
+Support for Melissa Climate A/C.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/climate.melissa/
+"""
+import logging
+
+from homeassistant.components.climate import ClimateDevice, \
+    SUPPORT_OPERATION_MODE, SUPPORT_TARGET_TEMPERATURE, SUPPORT_ON_OFF, \
+    STATE_AUTO, STATE_HEAT, STATE_COOL, STATE_DRY, STATE_FAN_ONLY, \
+    SUPPORT_FAN_MODE
+from homeassistant.components.fan import SPEED_LOW, SPEED_MEDIUM, SPEED_HIGH
+from homeassistant.components.melissa import DATA_MELISSA, DOMAIN
+from homeassistant.const import TEMP_CELSIUS, STATE_ON, STATE_OFF, \
+    STATE_UNKNOWN, STATE_IDLE, ATTR_TEMPERATURE, PRECISION_WHOLE
+
+DEPENDENCIES = [DOMAIN]
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_FLAGS = (SUPPORT_TARGET_TEMPERATURE | SUPPORT_OPERATION_MODE |
+                 SUPPORT_ON_OFF | SUPPORT_FAN_MODE)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
```

Comment:
Can these devices be discovered with the help of netdisco? If so, `discovery_info is not None` should be handled.

Reply:
they can not be discovered with netdisco unfortunately, that would have been great.

Reply:
Do you mean that netdisco is currently missing this or that it's not possible at all? Have you checked if they use SSDP/UPnP (this is probably easiest to catch with tcpdump/wireshark) or mDNS (`avahi-browse -ar`)?

Reply:
I tried to catch it with netdisco and it could not see it. Ill do a test with avahi

Reply:
I can not see it with avahi-browse -ar

Reply:
Did you look if you saw any broadcasts from the device? I created a simple script (requires you to install `upnpclient` package): https://gist.github.com/3bf36d76da44a0b4583e14b3fc77bc8a - that will print out information about SSDP/UPnP devices.

Note that the computer you run this has to be on the same network than the devices being discovered. Same applies for that `avahi-browse` command.

Reply:
No it's not found with script either.

Reply:
Ill test wiresharking for it

Reply:
Nope, the only way I can discover this device is through nmap (nowing the IP) and it actually have a basic auth on port 80 :D

Reply:
Ok, fair enough! I hope someone else will also take a look into this and merge it soon! :-)

Reply:
Thank you for the review! 🐬